editor/mainwindow.cpp

```diff
@@ -607,6 +607,34 @@ void MainWindow::onClientRowChanged(int row)
 		trackViews[i]->updateRow(row);
 }
 
+void MainWindow::onClientKeyChanged(int track, int row, float value, SyncTrack::TrackKey::KeyType type)
+{
+	if (track < doc->getTrackCount()) {
+		SyncTrack::TrackKey newKey;
+		SyncTrack *t = doc->getTrack(track);
+
+		/* ensure the editor is on the row being changed by the demo */
+		onClientRowChanged(row);
+
+		newKey.type = type;
+		newKey.row = row;
+		if (t->isKeyFrame(row))
+			newKey = t->getKeyFrame(row);
+
+		newKey.value = value;
+
+		doc->setKeyFrame(t, newKey);
+		onCurrValDirty();
+	} else
+		QApplication::beep();
```

Comment:
I don't think we should give a UI beep for the editor if the problem occurred in the other end. I get that there's no good way of reporting the error back to the client here, but I don't think this is a good alternative.